hsds/domain_sn.py

```diff
@@ -703,6 +703,23 @@ async def GET_Domain(request):
     app = request.app
     params = request.rel_url.query
 
+    search_root_object = None
+    include_links = False
+    include_attrs = False
+    follow_soft_links = False
+    follow_external_links = False
+
+    if "search_root_object" in params and params["search_root_object"]:
```

Comment:
This parameter was necessary for relative searches, because placing an object id directly into the request URL (e.g. `https://127.0.0.1:5101/g-111-222-333?h5path=child_group`) leads to a 404 error being raised before the service node is even reached. So instead that id would be used as a relative root like `https://127.0.0.1:5101/?h5path=child_group&search_root_object=g-111-222-333`

Reply:
Right, https://127.0.0.1:5101/g-111-222-333?h5path=child_group won't work as there's no route mapped to that in servicenode.py.

I'd stick with just the '/' path (i.e. https://127.0.0.1:5101/) and use the following logic in GET_Domain:

1.   No domain defined: return all top level domains (current logic)
2. h5path set in query param, but parent_id is not in query param:  return object given by the path
3. h5path and parent_id set in query param: return object in relative path from parent
4. h5path not set: Do normal domain GET

Reply:
If I'm understanding you correctly, I think this matches the current behavior, with `parent_id` defined by the `search_root_object`.